### PR TITLE
[SPARK-45419][SS] Avoid reusing rocksdb sst files in a dfferent rocksdb instance

### PR DESCRIPTION
**Pull request by chaoqin-li1123**
### What changes were proposed in this pull request?
When loading a rocksdb instance, remove file version map entry of larger versions to avoid rocksdb sst file unique id mismatch exception. The SST files in larger versions can't be reused even if they have the same size and name because they belong to another rocksdb instance.

### Why are the changes needed?
Avoid rocksdb file mismatch exception that may occur in runtime.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Add rocksdb unit test.

### CODE REVIEW

**Comment by chaoqin-li1123** [comment]
@anishshri-db @jerrypeng @siying

File: sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala [hunk 1]
```diff
@@ -240,6 +240,32 @@ class RocksDBSuite extends AlsoTestWithChangelogCheckpointingEnabled with Shared
     }
   }
 
+  testWithChangelogCheckpointingEnabled("Rollback multiple versions") {
```
Comment (anishshri-db) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala:243]: Can we give a more descriptive test name ? Also maybe add a comment linking the SPARK issue and just a small description of which case this test is validating

**Review by anishshri-db — APPROVED** [review]

**Comment by chaoqin-li1123** [comment]
@HeartSaVioR

**Review by HeartSaVioR — COMMENTED** [review]
The code change looks great. The PR description is a bit off though - would you mind restoring the template of PR description for OSS PR and fill it out?

**Review by HeartSaVioR — APPROVED** [review]
+1

**Comment by HeartSaVioR** [comment]
Thanks! Merging to master.

**Comment by HeartSaVioR** [comment]
Just ported the fix back to 3.5 as well.

**Pull request closed without merging**